test/unit/model/helpers/downloader-test.js

```diff
@@ -270,5 +270,15 @@ describe('Downloader', function() {
 
       expect(successHandler).to.be.calledOnce;
     });
+
+    it('should able to test restartDownload and downloadAuth', function() {
```

Comment:
You targeting restartDownload() function is this test. Then it should not be inside `describe('download', function(){` instead you should create 
``` JavaScript
describe('restartDownload', function() {
});
```
as a sibling for it to follow the pattern
* Class
  * method
     * should test 1
     * should test 2

``` JavaScript
...
describe('download', function() {
...
     it('should make a request with given options', function() {
...
     });
...
     it('should make a request with given url in options', function() {
...
     });
});

describe('restartDownload', function() {
    // create pending tests
});
```

then fill it with pending tests that would reads as specification for the function

``` JavaScript
describe('restartDownload', function() {
    it('should change downloader status from \'Download Failed\' to \'Downloading\'');
    it('should create new WriteStream instance for failed download entries');
    it('should call authDownload for entries that requires authentication');
    it('should call download method for entries that does not require authentication');
});
```

when you run unit tests you would see results report matching pattern defined above.
``` text
 Downloader
    √ should set totalDownloads to 1 by default
    √ responseHandler should set the total download size
    √ dataHandler should update the progress once time threshold is reached
    √ dataHandler should not update the progress before time threshold is reached
    √ errorHandler should close the stream
    √ endHandler should end the stream
    √ closeHandler should verify downloaded files checksum
    √ closeHandler should call success when verification succeeds
    √ closeHandler should call failure when verification fails
    download
      √ should make a request with given options
      √ should make a request with given url in options
      √ should make a request with 'User-Agent' header set
      √ should call endHandler when end event is emitted
      √ should call errorHandler when error event is emitted
      √ should save downloads in map
      √ should not call sucessHandler after error event is emitted
      √ should call sucessHandler ony after all downloads are finished
    restartDownload
      - should change downloader status from 'Download Failed' to 'Downloading'
      - should create new WriteStream instance for failed download entries
      - should call authDownload for entries that requires authentication
      - should call download method for entries that does not require authentication
```

then you implement every pending test to pass.